### ISSUE TITLE
Update test_banking_stage_entryfication to be more deterministic

### ISSUE THREAD

**Issue by t-neIson** [solana-labs-community/solana#201]
#### Problem
`test_banking_stage_entryfication` is flaky and goes into infinite loops sometimes.

#### Summary of Changes
The test sends two transactions and expects them to be part of two different entries. It was using `sleep` to induce delay in the posting of the transaction, so that the banking stage receive loop will process the two transaction in two separate iterations. But that's not very deterministic and causes the test to fail (and go in a loop).

This PR updates the test to ensure that the first transaction is recorded before the second is sent.

Fixes #

**Comment by sakeridge** [comment]
I think it's correct currently in the PR. The first transaction will bring the balance to 1. So it'll loop as long as the balance is < 1.

The second transaction will bring up the balance to 2. So the second loop waits till the balance reaches 2.

**Comment by CrieasofCarrots** [comment]
Really? The first transaction transfers 2 lamports and the second transaction transfers 1

**Comment by sakeridge** [comment]
Yes, you are right. I missed the transaction details. I'll update the test and push an update.

Thanks for catching it.
